### DISCUSSION TITLE
-

### DISCUSSION THREAD

**Discussion by DanRidh** [scikit-hep/awkward#2278] [Q&A]
I'm currently trying to write an element-wise product (Hadamard product) function but am facing issues when the input fed is of a weird nested format. It works fine when lets say the input is;
```
input_1: [1,2,3]
input_2: [4,5,6]
output: [4,10,18]
```

However, when there is nesting within a specific index i'm not sure how to get the expected output (flatenning and casting seem to be throwing errors, though this is probably due to a lack of deeper understanding on my end)

eg:
```
input_1: [[1], [2,[3]]]
input_2: [[4], [5,[6]]]
expected_output: [[4], [10,[18]]]
```

Any guidance would be really helpful!

**Comment by agoose77** [comment]
If I multiple the arrays as you give them, the element-wise operation just works:
```python
>>> x = ak.Array([[1], [2,[3]]])

>>> y = ak.Array( [[4], [5,[6]]])

>>> x * y
<Array [[4], [10, [18]]] type='2 * var * union[int64, var * int64]'>
```

Is this not what you observe?

**Reply by DanRidh** [reply]
@agoose77 thanks for such a quick response! When I try running the same as you posted it doesn't seem to create that expected output. It specifically throws an error `TypeError: concatenate() got an unexpected keyword argument 'casting'` for some reason

**Reply by agoose77** [reply]
Could you clarify the version of Awkward Array that you're running?
```python
>>> import awkward as ak
>>> ak.__version__
```

**Reply by DanRidh** [reply]
@agoose77 sure thing, I'm currently using version `2.0.8`

**Reply by agoose77** [reply]
Did you copy the program directly, and run it in a new session? I can't reproduce this on my end.

**Reply by DanRidh** [reply]
@agoose77 I just installed awkward through pip, opened a kernel on jupyter-notebook and imported awkward. Could it maybe be due to python version? I'm currently using `v3.8.12`

**Reply by agoose77** [reply]
Could you share the traceback? And, perhaps also the notebook?
